docs/rollup-boost-ha.md

```diff
@@ -0,0 +1,328 @@
+    CL->>RB: engine_forkChoiceUpdated(..., Attrs)
+    RB->>EL: engine_forkChoiceUpdated(..., Attrs)
+    RB->>B: engine_forkChoiceUpdated(..., Attrs)
+    B->>B: Start payload job
+
+    %% get_payload from builders
+    CL->>RB: engine_getPayload(PayloadId)
+    RB->>EL: engine_getPayload(PayloadId)
+    RB->>B: engine_getPayload(PayloadId)
+    B-->>RB: {executionPayload, blockValue}
+    EL-->>RB: {executionPayload, blockValue}
+
+    %% block selection policy
+    RB->>RB: Select best payload using block selection policy 
+
+    %% Validate best payload
+    Note over RB, EL: Validate builder block
+    RB->>EL: engine_newPayload(ExecutionPayload)
+    EL->>RB: {status: VALID, ...}
+
+    %% Propagate block to builders
+    Note over RB: Propagate new block to builders
+    RB->>B: engine_newPayload(ExecutionPayload)
+
+    %% Return final payload to op-node
+    RB->>CL: {executionPayload, blockValue}
+
+```
+
+This design ensures builder fault tolerance without compromising any fault tolerance/liveliness guarantees provided by the canonical HA sequencer setup. In an HA setup, a chain can run `n` builders for redundancy. If `n-1` builders go down, the remaining builder will still produce a block. If all builders are down, `rollup-boost` will fallback to the default execution client's block. This enables smooth upgrades and rolling deployments without disrupting external block production.
+
+The setup is flexible, allowing chains to decide how to provision their builder infrastructure however best fits their use case. Chains can run a single builder, `n` builders, a load-balanced set of builders behind a single builder endpoint, or multiple independent builders operated by different trusted parties. No coordination is required between builders, so operational overhead stays low as builder participation expands.
+
+Importantly, this design requires no changes to `op-conductor`, allowing Conductor simply query `op-node` for health checks, keeping builder health and state completely out of the leader election process.
+
+One potential issue with this design is that `op-conductor` does not strictly guarantee there will only be one active sequencer at a time. Specifically, there are edge cases where the `stopSequencer` call is not received in time (or if the call fails) by the previous leader’s `op-node` after a conductor failover. This edge case is acknowledged in the [op-conductor codebase and currently considered acceptable](https://github.com/ethereum-optimism/optimism/blob/273fd6abd4a92824e05f3daa9322b0c8add9e843/op-conductor/conductor/service.go#L766). The [Engine API](https://github.com/ethereum/execution-apis/blob/main/src/engine/common.md#load-balancing-and-advanced-configurations) spec states it is undefined behavior when multiple consensus clients concurrently control the head of a single execution client. Some clients may tolerate this, but it’s not standardized and could lead to unpredictable behavior.
```

Comment:
This is a "normal" case that HA system needs to handle. Think of the situation where conductor is down, leadership transfers and now we have 2 active sequencers. The system is able to handle this case to tolerate 2 active sequencers without affecting entire network's health

Reply:
The design doc has been updated to focus on 1:1 Rollup Boost <-> Builder deployments which ensures that there is only one `op-node` sending FCUs with payload attributes to the EL.